plugins/woo-ai/src/hooks/useStoreBranding.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from 'react';
+import { useQuery, UseQueryResult } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import { getToneOfVoice, getBusinessDescription } from '../utils/branding';
+
+// Define your data type
+interface BrandingData {
+	toneOfVoice: string;
+	businessDescription: string;
+}
+
+// Define your error type
+interface BrandingError {
+	message: string;
+}
+
+type UseStoreBrandingOptions = {
+	onError?: ( error: BrandingError ) => void;
+};
+
+// Async function to fetch branding data
+async function fetchBrandingData(): Promise< BrandingData > {
+	const toneOfVoice = await getToneOfVoice();
+	const businessDescription = await getBusinessDescription();
+
+	return { toneOfVoice, businessDescription };
+}
+
+export function useStoreBranding( {
+	onError,
+}: UseStoreBrandingOptions = {} ): UseQueryResult<
+	BrandingData,
+	BrandingError
+> {
+	const result = useQuery< BrandingData, BrandingError >(
```

Comment:
Having this directly in the `render` function will result it the query being run every time the hook is rendered (which could be often). This doesn't end up mattering too much due to the caching in react-query, but a more common approach would probably be to put the whole request into a `useEffect` with an empty dep array, and set the result to the state when it's complete.

Reply:
The problem is that `useQuery()` is itself a hook and thus can't be used within `useEffect()` 🤔